app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -124,6 +135,43 @@ def get_images
         process_collection(images, :vms) { |image| parse_image(image) }
       end
 
+      def get_security_groups
+        security_groups = gather_data_for_this_region(@nsg)
+
+        process_collection(security_groups, :security_groups) do |sg|
+          parse_security_group(sg)
+        end
+
+        get_firewall_rules
+      end
+
+      # To get the list of security groups for a VM we need to find any
+      # NIC's associated with the security group, and then get the VM's
+      # associated with the NIC's.
+      #
+      def get_security_groups_for_vm(vm)
+        group_array = []
+
+        security_groups.each do |sg|
+          sg.properties.network_interfaces.each do |nic|
+            resource_group = nic.id[%r{resourceGroups/(.*?)/}, 1]
+            nic = @nis.get(File.basename(nic.id), resource_group)
+            if nic.properties['virtualMachine'] and nic.properties.virtual_machine.id == vm.id
+              group_array << sg
+            end
+          end
+        end
+
+        group_array
```

Comment:
@blomquisg @djberg96 

Agree on the final question, I do not see why we need lines:
`instance.properties.network_profile.network_interfaces.each do |nic|
          pattern = %r{/subscriptions/(.+)/resourceGroups/([\w-]+)/.+/networkInterfaces/(.+)}i
          _m, sub, group, nic_name = nic.id.match(pattern).to_a
          cfg = @config.clone.tap { |c| c.subscription_id = sub }
          nic_profile = ::Azure::Armrest::Network::NetworkInterfaceService.new(cfg).get(nic_name, group)`

we can reuse a single Azure::Armrest::Network::NetworkInterfaceService instance variable